src/core/frontend/framebuffer_layout.cpp

```diff
@@ -39,115 +39,19 @@ static Common::Rectangle<T> MaxRectangle(Common::Rectangle<T> window_area,
-            // bottom
-            screen_window_area = {0, 0, width, top_screen.GetHeight()};
-            bot_screen = MaxRectangle(screen_window_area, BOT_SCREEN_ASPECT_RATIO);
-            bot_screen = bot_screen.TranslateX((top_screen.GetWidth() - bot_screen.GetWidth()) / 2);
-            if (swapped) {
-                bot_screen = bot_screen.TranslateY(height / 2 - bot_screen.GetHeight());
-            } else {
-                top_screen = top_screen.TranslateY(height / 2 - top_screen.GetHeight());
-            }
-        }
-    }
-    if (upright) {
-        // Move the top screen to the right if we are swapped.
-        res.top_screen = swapped ? top_screen.TranslateX(width / 2) : top_screen;
-        res.bottom_screen = swapped ? bot_screen : bot_screen.TranslateX(width / 2);
-    } else {
-        // Move the top screen to the bottom if we are swapped.
-        res.top_screen = swapped ? top_screen.TranslateY(height / 2) : top_screen;
-        res.bottom_screen = swapped ? bot_screen : bot_screen.TranslateY(height / 2);
-    }
-    return res;
+    return LargeFrameLayout(width, height, swapped, upright, 1.0f,
+                            Settings::SmallScreenPosition::BelowLarge);
 }
 
 FramebufferLayout PortraitTopFullFrameLayout(u32 width, u32 height, bool swapped) {
     ASSERT(width > 0);
     ASSERT(height > 0);
-
-    FramebufferLayout res{width, height, true, true, {}, {}, true, true};
-    // Default layout gives equal screen sizes to the top and bottom screen
-    Common::Rectangle<u32> screen_window_area{0, 0, width, height / 2};
-    Common::Rectangle<u32> top_screen = MaxRectangle(screen_window_area, TOP_SCREEN_ASPECT_RATIO);
-    Common::Rectangle<u32> bot_screen = MaxRectangle(screen_window_area, BOT_SCREEN_ASPECT_RATIO);
-
-    float window_aspect_ratio = static_cast<float>(height) / width;
-    // both screens height are taken into account by multiplying by 2
-    float emulation_aspect_ratio = TOP_SCREEN_ASPECT_RATIO * 2;
-
-    if (window_aspect_ratio < emulation_aspect_ratio) {
-        // Apply borders to the left and right sides of the window.
-        top_screen =
-            top_screen.TranslateX((screen_window_area.GetWidth() - top_screen.GetWidth()) / 2);
-        bot_screen =
-            bot_screen.TranslateX((screen_window_area.GetWidth() - bot_screen.GetWidth()) / 2);
-    } else {
-        // Window is narrower than the emulation content
-        // Recalculate the bottom screen to account for the width difference between top and bottom
-
-        bot_screen = bot_screen.TranslateX((top_screen.GetWidth() - bot_screen.GetWidth()) / 2);
-    }
-
-    // Move the top screen to the bottom if we are swapped.
-    res.top_screen = swapped ? top_screen.TranslateY(bot_screen.GetHeight()) : top_screen;
-    res.bottom_screen = swapped ? bot_screen : bot_screen.TranslateY(top_screen.GetHeight());
-
+    float sf = swapped ? 1.25f : 0.8f;
```

Comment:
What does "sf" stand for here? Whatever it is should probably be the name of the value.
Also, make this a `const`.

Reply:
it is `scale_factor`, I'll fix.